server/src/controllers/patchProfile.js

```diff
@@ -0,0 +1,32 @@
+const Boom = require('@hapi/boom');
+const bcrypt = require('bcrypt');
+const patchProfile = require('../database/queries/patchProfile');
+const profileScema = require('./validation/profileSchema');
+const getUserById = require('../database/queries/getUserById');
+
+module.exports = (req, res, next) => {
+  const { userID } = req.user;
+  const { name, oldPassword = '', password } = req.body;
+
+  profileScema
+    .validateAsync({ name, password }, { abortEarly: false })
+    .then(() => getUserById(userID))
+    .then((users) => {
+      if (users.rows.length === 0) {
+        throw new Error('user does not exists');
```

Comment:
use boom errors, this would lead to 404 not found

Reply:
I have used `boom.notFound` here, but the server respond with 400 badRequest instead
I don't know why,
using `throw new Error` here will skip to
```js
.catch((error) => {
      next(Boom.badRequest(error));
```
its the same thing